node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -1152,26 +1157,61 @@ mod tests {
             from: wallet,
             wei_amount: 1_000_000_000,
         };
-        let all_received_payments = vec![suspect.clone()];
         let txn = conn.transaction().unwrap();
         let logger = Logger::new(test_name);
 
-        let result = ReceivableDaoReal::verify_possibly_unknown_wallet(
-            txn.as_ref(),
-            &logger,
-            &suspect,
-            &all_received_payments,
-        );
+        let result = ReceivableDaoReal::verify_possibly_unknown_wallet(&txn, &logger, &suspect);
 
         let expected_panic_msg = "Update for received payment with 1000000000 wei ran \
-        without producing a data change, despite a record for the wallet \
-        0x00000000000000000000000000000000626c6168 exists";
-        assert_eq!(
-            result,
-            Err(ReceivableDaoError::RusqliteError(
-                expected_panic_msg.to_string()
-            ))
-        )
+        without producing a change in the database, despite the record for wallet \
+        0x00000000000000000000000000000000626c6168 exists.";
+        assert_eq!(result, Err(RusqliteError(expected_panic_msg.to_string())))
+    }
+
+    #[test]
+    fn verify_possibly_unknown_wallet_returns_error_finding_duplicated_records_for_single_wallet() {
```

Comment:
This test tests an impossible condition: that there are several records in the table with the same wallet address.

The actual definition for this table, from `db_initializer.rs`, is:
```
create table if not exists receivable (
        wallet_address text primary key,
        balance_high_b integer not null,
        balance_low_b integer not null,
        last_received_timestamp integer not null
) strict
```
Notice that the `wallet_address` column is a primary key; so uniqueness will be enforced. Also notice that the balance column is actually two columns, both of which are `not null`, and there's a `last_received_timestamp` column that's also `not null` which doesn't appear in the test table. I don't know whether there's an important difference between the `text` and `string` data types.

It's important to test with the same table schema you have in production. It might even be worth putting the table definitions in manifest constants and using the same constant in both places.